WordPress/gutenberg
# Query Loop: Add Page Order (menu_order)

## What?
This PR adds the possibility to the query loop block to output pages according to the order set in the backend (menu_order). For this purpose a new option is added to the existing dropdown menu. Since by default only pages have this sorting option, this option is only visible if Page is selected under Post Type.

## Why?
The current sorting options by creation date and title are not sufficient for most use cases. To build overview pages that automatically display and link to the appropriate subpages or specific pages, you often need to set an order that is not alphabetical.

According to this list, this was probably on the agenda anyway: https://github.com/WordPress/gutenberg/issues/24934

## How?
I have added a new option to the OrderControl component. The array orderOptions, which was static before, now has to be in the function, because we want to change it depending on the PostType and only show it when the PostType is Page.

In the parent component we extend the call of the OrderControl by the variable postType, so that the currently selected PostType is always passed to the OrderControl.

If the PostType is changed to Post or something other than Page, we have to reset the value to the default, because our new option is not available.

For this to work, we need to allow the menu_order option in the enum of the corresponding REST API call.

Theoretically, it would be even better if we don't check for postType === page, but check if the selected postType supports page-attributes. Then this option would also appear for CPTs that support this sorting. But since this is my first PR, I wanted to keep it as simple as possible for now.

## Testing Instructions

1. create several pages and sort them manually in the page list
2. create a query loop block and configure it so that the post type Page is selected
3. change the Order by option to the new Page Order option
4. verify that the pages are sorted in the previously set order.
5. change the Post Type option to Post and verify that the Order by option changes to Newest to oldest and the new Page order option is no longer displayed.
6. verify that these changes to the order are also identical on the frontend

## Screenshots or screencast
![Bildschirmfoto 2023-06-07 um 13 21 21](https://github.com/WordPress/gutenberg/assets/4995134/99b99e3d-b798-420e-ab47-303b5a1372a4)
![Bildschirmfoto 2023-06-07 um 13 21 07](https://github.com/WordPress/gutenberg/assets/4995134/06740826-62db-4e79-b556-ff19f9e9d5af)
